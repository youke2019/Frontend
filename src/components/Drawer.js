import React from "react";
import { View } from "react-native";
import {Overlay} from 'react-native-elements'
import FilterItem from './FilterItem'

class Drawer extends React.Component {
    state = {
        list:{
            credits:[
                {name:'1',value:'1'},
                {name:'2',value:'2'},
                {name:'3',value:'3'},
                {name:'4',value:'4'},
            ],
            general:[
                {name:'人文学科',value:'人文学科'},
                {name:'社会科学',value:'社会科学'},
                {name:'自然科学',value:'自然科学'},
                {name:'工程科学与技术',value:'工程科学与技术'},
            ],
            time:[
                {name:'周一',value:1},
                {name:'周二',value:2},
                {name:'周三',value:3},
                {name:'周四',value:4},
                {name:'周五',value:5},
                {name:'周六',value:6},
                {name:'周日',value:7},
            ]
        }
    }

    updateFilter = (filter) => {
        this.props.updateFilter(filter)
    }

    render() {
        const {
            visible = false,
            onBackdropPress=() => {},
        } = this.props

        const {list} = this.state

        return (
            <Overlay
                isVisible={visible}
                onBackdropPress={onBackdropPress}
            >
                <View>
                    <FilterItem
                        title="学分"
                        list={list.credits}
                        updateFilter={(filter) => this.updateFilter(filter)}
                    />
                    <FilterItem
                        title="通识类型"
                        list={list.general}
                        updateFilter={(filter) => this.updateFilter(filter)}
                    />
                    <FilterItem
                        title="上课时间"
                        list={list.time}
                        updateFilter={(filter) => this.updateFilter(filter)}
                    />
                </View>
            </Overlay>
        );
    }
}

export default Drawer