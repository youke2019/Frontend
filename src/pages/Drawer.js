import React from "react";
import {Image, Text, View} from "react-native";

class Drawer extends React.Component {
    render() {
        return (
            <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
                <Text>Home!</Text>
                <Image source={{uri: 'building'}} style={{width: 40, height: 40}} />
            </View>
        );
    }
}

export default Drawer