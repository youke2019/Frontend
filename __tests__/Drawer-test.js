import React from 'react';
import Drawer from '../src/components/Drawer';

import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16'
import {shallow, mount, render, configure} from 'enzyme'

configure({adapter: new Adapter()})

const updateFilter = jest.fn((filter) => {

})

test('renders Drawer correctly', () => {
    const tree = renderer.create(<Drawer/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('test function updateFilter()', () => {
    let wrapper = shallow(<Drawer updateFilter={updateFilter}/>)

    wrapper.find({title:'学分'}).props().updateFilter()
    wrapper.find({title:'通识类型'}).props().updateFilter()
    wrapper.find({title:'上课时间'}).props().updateFilter()
    expect(updateFilter).toHaveBeenCalledTimes(3)
});

