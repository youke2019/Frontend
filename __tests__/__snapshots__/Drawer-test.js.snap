// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Drawer correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "height": 400,
          "padding": 16,
        }
      }
    >
      <RCTScrollView
        style={Object {}}
      >
        <View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 8,
                    "fontSize": 16,
                  }
                }
              >
                学分
              </Text>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "arrow_down",
                    }
                  }
                  style={
                    Object {
                      "height": 21,
                      "width": 21,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 8,
                    "fontSize": 16,
                  }
                }
              >
                通识类型
              </Text>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "arrow_down",
                    }
                  }
                  style={
                    Object {
                      "height": 21,
                      "width": 21,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 8,
                    "fontSize": 16,
                  }
                }
              >
                上课时间
              </Text>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "arrow_down",
                    }
                  }
                  style={
                    Object {
                      "height": 21,
                      "width": 21,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 8,
                    "fontSize": 16,
                  }
                }
              >
                上课地点
              </Text>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "arrow_down",
                    }
                  }
                  style={
                    Object {
                      "height": 21,
                      "width": 21,
                    }
                  }
                />
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 8,
                    "fontSize": 16,
                  }
                }
              >
                开设学院
              </Text>
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "arrow_down",
                    }
                  }
                  style={
                    Object {
                      "height": 21,
                      "width": 21,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FDD32A",
              "borderRadius": 20,
              "height": 30,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginHorizontal": 6,
              "marginTop": 18,
              "opacity": 1,
              "width": 50,
            }
          }
        >
          <Text>
            确定
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FDD32A",
              "borderRadius": 20,
              "height": 30,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginHorizontal": 6,
              "marginTop": 18,
              "opacity": 1,
              "width": 50,
            }
          }
        >
          <Text>
            取消
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
